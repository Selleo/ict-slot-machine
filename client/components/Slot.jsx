import React from 'react'

class Slot extends React.Component {
  render() {
    return (
      <div style={this.props.style} className='slot'>
        <img src={ require(`../images/icon-${this.props.number + 1}.png`) } />
      </div>
    )
  }
}

export default Slot
