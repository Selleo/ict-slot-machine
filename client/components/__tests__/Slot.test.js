import React from 'react'
import renderer from 'react-test-renderer'
import Slot from '../Slot'

describe('<Slot />', () => {
  const defaultProps = {
    polygon: {
      valuesFor: () => ({
        translateZ: 10,
        translateY: 10,
        rotateX: 10
      })
    },
    number: 0
  }
  const wrapper = renderer.create(<Slot {...defaultProps} />)

  test('render', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
