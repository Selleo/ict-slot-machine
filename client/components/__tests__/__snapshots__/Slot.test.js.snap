// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slot /> render 1`] = `
<div
  className="slot"
  style={
    Object {
      "transform": "translateZ(10px) translateY(10px) rotateX(10deg)",
    }
  }
>
  <img
    src="test-file-stub"
  />
</div>
`;
